### ISSUE TITLE
Close button doesn't work when contains any elements

### ISSUE THREAD

**Issue by DimaGashko** [ghosh/Micromodal#392]
I've added an svg icon to the close button and realized that the button doesn't work anymore.

Reason: you use event delegation slightly incorrect (https://github.com/ghosh/Micromodal/blob/master/lib/src/index.js#L131).

```js
if (event.target.hasAttribute(this.config.closeTrigger)) {
    this.closeModal(event)
}
```

Solution: replace `hasAttribute`  with  `closest`:
```js
if (event.target.closest(`[${this.config.closeTrigger}]`)) ...
```

Html:
 ```html
 <button data-micromodal-close>
        <svg><use xlink:href="#close.svg"></use></svg>
 </button>
```

p.s. For now I'm just going to use something like:
 ```css
.modal__close svg {
    pointer-events: none;
}
```
It's working, but I hope you would  fix it. 

Details: https://javascript.info/event-delegation

**Comment by deniseismo** [comment]
I can confirm this: had the same issue where the close button with an svg icon inside wouldn't close properly (occasionally it did work), thought I was tripping. Thanks for the workaround.

**Comment by HongPong** [comment]
for eslint i put
```
if (event.target.closest(`[${ this.config.closeTrigger }]`)) {
```
seems to work as it should. thank you!

**Comment by ghosh** [comment]
Fixed in #443. Thanks @HongPong
